packages/components/src/color-picker/test/index.tsx

```diff
@@ -73,99 +38,188 @@ const legacyColorMatcher = {
 describe( 'ColorPicker', () => {
 	describe( 'legacy props', () => {
 		it( 'should fire onChangeComplete with the legacy color format', async () => {
+			const user = userEvent.setup();
 			const onChangeComplete = jest.fn();
-			const color = '#fff';
+			const color = '#000';
 
-			const { container } = render(
+			render(
 				<ColorPicker
 					onChangeComplete={ onChangeComplete }
 					color={ color }
+					enableAlpha={ false }
 				/>
 			);
 
-			const saturation = getSaturation( container );
+			const formatSelector = screen.getByRole( 'combobox' );
 
-			if ( saturation === null ) {
-				throw new Error( 'The saturation slider could not be found' );
+			if ( formatSelector === null ) {
+				throw new Error(
+					'The color format selector could not be found'
+				);
 			}
 
-			expect( saturation ).toBeInTheDocument();
+			expect( formatSelector ).toBeInTheDocument();
 
-			moveReactColorfulSlider(
-				saturation,
-				{ pageX: 0, pageY: 0 },
-				{ pageX: 10, pageY: 10 }
-			);
+			await user.selectOptions( formatSelector, 'hex' );
 
-			await waitFor( () =>
-				expect( onChangeComplete ).toHaveBeenCalled()
-			);
+			const hexInput = screen.getByRole( 'textbox' );
 
-			expect( onChangeComplete ).toHaveBeenCalledWith(
+			if ( hexInput === null ) {
+				throw new Error(
+					'The color format selector could not be found'
+				);
+			}
+
+			expect( hexInput ).toBeInTheDocument();
```

Comment:
Same as above. Isn't the assertion enough?

One little additional nit here is that I find `.toBeVisible()` more specific than `.toBeInTheDocument()` since `.toBeInTheDocument()` may be true but the user won't see the element if it's unintentionally hidden.

Reply:
Also, `.toBeVisible` is testing something that we don't know yet. If a `screen.getByRole` query function returns an element, then that element is always in the document, isn't it? Where else it would be?

`.toBeInTheDocument` checks can be useful when it's the only thing that we're checking in the test. A test without any `expect()` call would trigger a lint error.

Reply:
👍